modules/quest/src/main/java/unnamed/mmo/quest/QuestProgress.java

```diff
@@ -0,0 +1,7 @@
+package unnamed.mmo.quest;
+
+// An enum list meant to contain ids for major story events to act as flags for the quest system
+public enum QuestProgress {
```

Comment:
Should not exist IMO. I am generally against hardcoding anything that is content related. In this case, if someone writing quests wants to add a new "main" quest line quest, they need to add it in here, rather than just editing some data file.

Reply:
Alright, though I would want some sort of list that is checked, but does not fail against so that we can catch misspelling early, and we can just have questprogress be an arbitrary string

Reply:
Eventually we will have a tool to make writing the data files a bit easier. eg write them split into different files and directories, but then the tool joins them all. Also the item format kinda sucks, and a bunch of it can be determined automatically (eg assign ids in order so you dont have to write them). 
That tool can also have a validation step, this can be done there.